pkg/tfbridge/detailed_diff_test.go

```diff
@@ -42,11 +42,9 @@ func TestSchemaLookupMaxItemsOnePlain(t *testing.T) {
 		},
 	}
 
-	differ := detailedDiffer{
-		tfs: shimv2.NewSchemaMap(sdkv2Schema),
-	}
+	tfs := shimv2.NewSchemaMap(sdkv2Schema)
```

Comment:
The changes to tests look very unconvincing 🙏 

You're going for no change in behavior in this PR? 

Can you point me to a test case that illustrates this "short-circuit" diff so I can build up some intuition? Perhaps this all is under tests already. Thank you!

Do not want to block on this but would like to understand better.

Reply:
The refactor moved the lookupSchemas function to be independent of the differ class, hence these changes.

Which part is unconvincing? The behaviour is meant to be under tests, hence why there are no additional changes here - the existing tests still work fine.


The short-circuiting behaviour is triggered when the two propertyValue trees we are diffing are not balanced - e.g. we see a nil versus a collection or a nil vs an object. Previously we'd recurse into the object and compare its properties against nil but now we short-circuit on encountering a nil. This enabled the whole thing to work for sets where we can not recurse into set elements.

Reply:
The change is large and I'm pretty sure changes behavior. I don't think this is a refactor given the test cov we have.

Reply:
The change is big but I have added quite a few tests here, pretty sure the coverage is good. The follow-up work also works quite well, so I am fairly confident in this change.

What do you think is missing?